### PR TITLE
rgw: fix unhandled switch case for AMQP_STATUS_SSL_SET_ENGINE_FAILED

### PR DESCRIPTION
**Pull request by sisco0**
rgw: fix unhandled switch case for AMQP_STATUS_SSL_SET_ENGINE_FAILED

Fixes a compiler warning referenced in the ticket issue.

Fixes: [https://tracker.ceph.com/issues/53252](https://tracker.ceph.com/issues/53252)
Signed-off-by: Francisco J. Solis <siscomagma@gmail.com>

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests
- Teuthology
  - [x] Completed teuthology run
  - [ ] No teuthology test necessary (e.g., documentation)

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by sisco0** [comment]
The CI build fails as the _RabbitMQ_ version is not compared against a version that currently provides the introduced enumerator value. https://github.com/alanxz/rabbitmq-c/commit/f182c13765e2a653da3bd9c3b49cac92e8dbd9fa is tagged as included in v0.11.0, as `apt` seems to provide v0.10.0 as shown below:

```bash
❯ sudo apt-cache policy librabbitmq-dev
librabbitmq-dev:
  Installed: 0.10.0-1
  Candidate: 0.10.0-1
  Version table:
 *** 0.10.0-1 500
        500 http://mx.archive.ubuntu.com/ubuntu groovy/main amd64 Packages
        100 /var/lib/dpkg/status
```

We should set a conditional pragma for the case of these versions and up, taking into account the definitions shown in the `amqp.h` file.
https://github.com/alanxz/rabbitmq-c/blob/219efb1179d9755b243adf69519a1344433bd964/librabbitmq/amqp.h#L221-L224

File: src/rgw/rgw_amqp.cc [hunk 1]
```diff
@@ -322,6 +322,10 @@ std::string to_string(amqp_status_enum s) {
       return "AMQP_STATUS_SSL_CONNECTION_FAILED";
     case _AMQP_STATUS_SSL_NEXT_VALUE:
       return "AMQP_STATUS_INTERNAL"; 
+#if AMQP_VERSION_MAJOR >= 0 && AMQP_VERSION_MINOR >= 11
```
Comment (yuvalif) [inline src/rgw/rgw_amqp.cc:325]: please write the condition similarly to line 303

Comment (sisco0) [reply src/rgw/rgw_amqp.cc]: Thank you for the given commit, the line has been modified accordingly.

**Review by yuvalif — APPROVED** [review]

**Comment by sisco0** [comment]
Closed in favor of https://github.com/ceph/ceph/pull/48238, as current PR does not contain sign information in the latest commit.

**Comment by yuvalif** [comment]
> Closed in favor of #48238, as current PR does not contain sign information in the latest commit.

@sisco0 approved the other PR. 
BTW, you can also do `git commit --amend` and then `git push --force` to fix the signing issue without switching branches

**Pull request closed without merging**